### PR TITLE
fix: [CDS-83545]: fix create agent when agent deleted outside of terraform

### PR DESCRIPTION
**Pull request by mteodor**
## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

**Review by mankrit-singh — APPROVED** [review]

File: internal/service/platform/gitops/deployment_spec/data_source_gitops_agent_deploy_yaml.go [hunk 1]
```diff
@@ -1,83 +0,0 @@
-package agent_yaml
```
[the remark this answers is not in the document]
Comment (mteodor) [reply internal/service/platform/gitops/deployment_spec/data_source_gitops_agent_deploy_yaml.go:1]: this code is moved into https://github.com/harness/terraform-provider-harness/blob/main/internal/service/platform/gitops/agent_yaml/data_source_gitops_agent_deploy_yaml.go, docs updated accordingly

**Review by mankrit-singh — APPROVED** [review]

**Comment by mteodor** [comment]
trigger build

File: docs/data-sources/platform_gitops_agent_deploy_yaml.md [hunk 2]
```diff
@@ -33,10 +33,22 @@ data "harness_platform_gitops_agent_deploy_yaml" "example" {
...
 
 - `id` (String) The ID of this resource.
 - `yaml` (String) The deployment manifest YAML of the GitOps agent.
+
+<a id="nestedblock--proxy"></a>
+### Nested Schema for `proxy`
+
+Optional:
+
+- `http` (String) HTTP proxy settings for the GitOps agent.
```
Comment (manavjot-harness) [inline docs/data-sources/platform_gitops_agent_deploy_yaml.md:51]: i suppose these changes are already implemented in the proxy request changes in the deploy yaml datasource?

Comment (mteodor) [reply docs/data-sources/platform_gitops_agent_deploy_yaml.md:51]: yes, implemented here https://github.com/harness/terraform-provider-harness/blob/main/internal/service/platform/gitops/agent_yaml/data_source_gitops_agent_deploy_yaml.go

**Review by manavjot-harness — APPROVED** [review]

**Comment by mankrit-singh** [comment]
trigger build

**Review by mankrit-singh — APPROVED** [review]

File: internal/service/platform/gitops/agent/data_source_gitops_agent.go [hunk 3]
```diff
@@ -111,13 +110,13 @@ func dataSourceGitopsAgentRead(ctx context.Context, d *schema.ResourceData, meta
...
-	if err != nil {
+	if err != nil && httpResp.StatusCode != 404 {
 		return helpers.HandleApiError(err, d, httpResp)
 	}
 
 	// Soft delete lookup error handling
 	// https://harness.atlassian.net/browse/PL-23765
-	if &resp == nil {
+	if &resp == nil || httpResp.StatusCode == 404 {
 		d.SetId("")
```
Comment (rathodmeetsatish) [inline internal/service/platform/gitops/agent/data_source_gitops_agent.go:120]: we should not do this in data source its needed in resource's read context only

File: go.mod [hunk 4]
```diff
@@ -20,15 +20,10 @@ require (
 
 require (
 	github.com/Azure/go-ansiterm v0.0.0-20210617225240-d185dfc1b5a1 // indirect
-	github.com/Masterminds/goutils v1.1.1 // indirect
```
Comment (rathodmeetsatish) [inline go.mod:23]: i guess only go sdk version change was needed please revert other library changes

**Comment by rathodmeetsatish** [comment]
Add test similar to TestAccResourcePipeline_DeleteUnderlyingResource to test out this scenario

**Pull request merged**